From fb2cd653896619a2bb2b494b70e8e9edfdef06fb Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 15 Nov 2011 17:12:50 +0100
Subject: [PATCH] dracut: add install_items to dracut.conf

install_items+=" <file>[ <file> ...] "

    Specify additional files to include in the initramfs, separated by
    spaces.

Backport of commit 0ec40e08d5649cc3f42d1b8af1ac49a28d1e6765
---
 dracut            |    9 +++++++--
 dracut.conf.5.xml |    8 ++++++++
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/dracut b/dracut
index ddafdef..c196e8e 100755
--- a/dracut
+++ b/dracut
@@ -202,13 +202,12 @@ while (($# > 0)); do
         -o|--omit)     push_arg omit_dracutmodules_l "$@" || shift;;
         -d|--drivers)  push_arg drivers_l            "$@" || shift;;
         --filesystems) push_arg filesystems_l        "$@" || shift;;
-        -I|--install)  push_arg install_items        "$@" || shift;;
+        -I|--install)  push_arg install_items_l      "$@" || shift;;
         --fwdir)       push_arg fw_dir_l             "$@" || shift;;
         -k|--kmoddir)  read_arg drivers_dir_l        "$@" || shift;;
         -c|--conf)     read_arg conffile             "$@" || shift;;
         --confdir)     read_arg confdir              "$@" || shift;;
         -L|--stdlog)   read_arg stdloglvl_l          "$@" || shift;;
-        -I|--install)  read_arg install_items        "$@" || shift;;
         --fwdir)       read_arg fw_dir_l             "$@" || shift;;
         --compress)    read_arg compress_l           "$@" || shift;;
         --prefix)      read_arg prefix_l             "$@" || shift;;
@@ -323,6 +322,12 @@ if (( ${#add_drivers_l[@]} )); then
     done
 fi
 
+if (( ${#install_items_l[@]} )); then
+    while pop install_items_l val; do
+        push install_items $val
+    done
+fi
+
 # these options override the stuff in the config file
 if (( ${#dracutmodules_l[@]} )); then
     dracutmodules=''
diff --git a/dracut.conf.5.xml b/dracut.conf.5.xml
index 697e655..2e08eab 100644
--- a/dracut.conf.5.xml
+++ b/dracut.conf.5.xml
@@ -112,6 +112,14 @@ initramfs.</para>
       </varlistentry>
       <varlistentry>
         <term>
+          <envar>install_items+=&quot;&#160;<replaceable>&lt;file&gt;</replaceable>[ <replaceable>&lt;file&gt;</replaceable>&nbsp;...]&nbsp;&quot;</envar>
+        </term>
+        <listitem>
+          <para>Specify additional files to include in the initramfs, separated by spaces.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term>
           <envar>do_strip=&quot;<replaceable>{yes|no}</replaceable>&quot;</envar>
         </term>
         <listitem>
